Assets/Scripts/InventoryAndItemsScripts/Shop/ShopController.cs

```diff
@@ -0,0 +1,266 @@
+        InventoryAction.performed -= OpenInventory;
+        InventoryAction.Disable();
+        SelectItem.performed -= BuyItem;
+        SelectItem.Disable();
+
+        // Navigating inventory
+        NavLeft.performed -= NavInvLeft;
+        NavRight.performed -= NavInvRight;
+        NavUp.performed -= NavInvUp;
+        NavDown.performed -= NavInvDown;
+
+        NavLeft.Disable();
+        NavRight.Disable();
+        NavUp.Disable();
+        NavDown.Disable();
+    }
+
+    void OpenInventory(CallbackContext c)
+    {
+        if (invObj.activeSelf)
+        {
+            GameStateManager.Instance.SetState(new PlayingState());
+        }
+        else
+        {
+            // make sure something else isn't already pausing the game
+            var state = GameStateManager.Instance.CurrentState;
+            if (state is not PauseState && state is not InventoryState) {
+                GameStateManager.Instance.SetState(new GameStateManagement.ShopState(this));
+            }
+        }
+    }
+
+    #region Inventory Navigation
+    void NavInvLeft(CallbackContext c)
+    {
+        if (invObj.activeSelf)
+        {
+            if (inventoryIndex > 0 && inventoryIndex % inventoryRowSize != 0) 
+            {
+                inventoryIndex--;
+                selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+                selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+                Debug.Log(inventoryIndex);
+            }
+        }
+    }
+
+    void NavInvRight(CallbackContext c)
+    {
+        if (invObj.activeSelf)
+        {
+            if (inventoryIndex < inventorySlots.Count && inventoryIndex % inventoryRowSize != inventoryRowSize - 1)
+            {
+                inventoryIndex++;
+                selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+                selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+                Debug.Log(inventoryIndex);
+            }
+        }
+    }
+
+    void NavInvUp(CallbackContext c)
+    {
+        if (invObj.activeSelf)
+        {
+            if (inventoryIndex > inventoryRowSize - 1) inventoryIndex -= inventoryRowSize;
+            selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+            selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+            Debug.Log(inventoryIndex);
+        }
+    }
+
+    void NavInvDown(CallbackContext c)
+    {
+        if (invObj.activeSelf)
+        {
+            if (inventoryIndex < inventorySlots.Count - inventoryRowSize) inventoryIndex += inventoryRowSize;
+            selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+            selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+            Debug.Log(inventoryIndex);
+        }
+    }
+    #endregion
```

Comment:
seems to be duplicate code from InventoryController, may be good to factor out both to some other class